independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -268,9 +268,11 @@ public Enumeration<URL> getResources(String unsanitisedName, boolean parentAlrea
         } else if (name.isEmpty()) {
             for (ClassPathElement i : elements) {
                 List<ClassPathResource> resList = i.getResources("");
-                for (var res : resList) {
-                    if (res != null) {
-                        resources.add(res.getUrl());
+                if (resList != null) {
```

Comment:
In this case while the fix is probably fine, it would be better to track down which implementations of `List<ClassPathResource> getResources(String name)` actually return null, and have them return an empty list.

I see the implementation in `PathTreeClassPathElement` does that - wouldn't you rather patch the implementation?

It feels wrong to me in this case as it could more safely return an empty list.